### PR TITLE
[#168247105] At the end of onboarding process, add a screen containing user's email validation CTA

### PR DESCRIPTION
**Pull request by lmorelli986**
This PR aims to add a screen containing user's email validation CTA at the end of onboarding process.

Schermata 2019-09-20 alle 17 55 49
===

Simulator Screen Shot - iPhone X - 2019-09-20 at 18 10 04

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#168247105](https://www.pivotaltracker.com/story/show/168247105): Come CIT al termine dell'onboarding, dopo aver inserito il mio indirizzo email, voglio vedere uno screen che mi ricorda di validarlo

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: locales/it/index.yml [hunk 1]
```diff
@@ -309,6 +309,21 @@ onboarding:
         faceId: FaceID
         touchId: TouchID
       notEnrolledText: !include tos_fingerprint_not_enrolled.md
+  email:
+    validation:
+      title: Ricorda di validare il tuo indirizzo email
+      headerTitle: Validazione email
+      info: "Ti abbiamo inviato una mail a <b>{{email}}</b> con un link per validare la mail:
+        clicca semplicemente su quel link.
+        Finché non validaerai il tuo indirizzo email alcune funzioni dell'app (pagamenti,
```
Comment (Undermaken) [inline locales/it/index.yml]:
```suggestion
        Finché non validerai il tuo indirizzo email alcune funzioni dell'app (pagamenti,
```

File: ts/sagas/startup/checkAcknowledgedEmailSaga.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * Launch email saga that consists of:
+ * - acknowledgement screen if email already exists and is valid and the user
+ *   does not want to change it
+ * - editing and validation screen, otherwise
+ */
+export function* checkAcknowledgedEmailSaga(
+  userProfile: UserProfileUnion
+): IterableIterator<Effect> {
+  // Check if user profile has email
+  if (userProfile.spid_email && userProfile.spid_email !== "") {
```
Comment (Undermaken) [inline ts/sagas/startup/checkAcknowledgedEmailSaga.ts]: Some of here https://github.com/teamdigitale/io-app/pull/1308#discussion_r327025006

File: locales/en/index.yml [hunk 3]
```diff
@@ -303,6 +303,21 @@ onboarding:
         faceId: FaceID
         touchId: TouchID
       notEnrolledText: !include tos_fingerprint_not_enrolled.md
+  email:
+    validation:
+      title: Remember to validate your email address
+      headerTitle: Email validation
+      info: "We sent an email to <b>{{email}}</b> with a link to validate your email:
+        click simply on that link.
```
Comment (Undermaken) [inline locales/en/index.yml]:
```suggestion
        click simply on that link.<br/>
```

File: locales/it/index.yml [hunk 4]
```diff
@@ -309,6 +309,21 @@ onboarding:
         faceId: FaceID
         touchId: TouchID
       notEnrolledText: !include tos_fingerprint_not_enrolled.md
+  email:
+    validation:
+      title: Ricorda di validare il tuo indirizzo email
+      headerTitle: Validazione email
+      info: "Ti abbiamo inviato una mail a <b>{{email}}</b> con un link per validare la mail:
+        clicca semplicemente su quel link.
```
Comment (Undermaken) [inline locales/it/index.yml]:
```suggestion
        clicca semplicemente su quel link.<br/>
```

File: ts/screens/onboarding/EmailValidateScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+const styles = StyleSheet.create({
+  spacerLarge: { height: 40 }
+});
+
+const unavailableAlert = () => Alert.alert(I18n.t("global.notImplemented"));
+
+/**
+ * A screen to show if the fingerprint is supported to the user.
+ */
+export class FingerprintScreen extends React.PureComponent<Props> {
```
Comment (Undermaken) [inline ts/screens/onboarding/EmailValidateScreen.tsx]:
```suggestion
export class EmailValidateScreen extends React.PureComponent<Props> {
```

File: ts/screens/onboarding/EmailValidateScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+import I18n from "../../i18n";
+import { BiometrySimpleType } from "../../sagas/startup/checkAcknowledgedFingerprintSaga";
+import {
+  abortOnboarding,
+  emailAcknowledged
+} from "../../store/actions/onboarding";
+import { Dispatch, ReduxProps } from "../../store/actions/types";
+import { GlobalState } from "../../store/reducers/types";
+
+type NavigationParams = {
```
Comment (Undermaken) [inline ts/screens/onboarding/EmailValidateScreen.tsx]: It is never used

File: ts/screens/onboarding/EmailValidateScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+            onPress: this.props.acknowledgeEmail
+          }}
+        />
+      </TopScreenComponent>
+    );
+  }
+}
+
+const mapStateToProps = (state: GlobalState) => ({
+  potProfile: pot.toOption(state.profile)
```
Comment (Undermaken) [inline ts/screens/onboarding/EmailValidateScreen.tsx]: same of here https://github.com/teamdigitale/io-app/pull/1308#discussion_r327025800

File: ts/screens/onboarding/EmailValidateScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+          style: "default",
+          onPress: () => this.props.abortOnboarding()
+        }
+      ]
+    );
+
+  public render() {
+    const { potProfile } = this.props;
+
+    const profileData = potProfile
```
Comment (Undermaken) [inline ts/screens/onboarding/EmailValidateScreen.tsx]: Could we have email directly?= https://github.com/teamdigitale/io-app/pull/1308#discussion_r327027072

File: ts/screens/onboarding/EmailValidateScreen.tsx (hunk 5 above)
Comment (lmorelli986) [reply ts/screens/onboarding/EmailValidateScreen.tsx]: same as #1308

File: locales/it/index.yml (hunk 1 above)
Comment (lmorelli986) [reply locales/it/index.yml]: changed in db325d10

File: locales/en/index.yml (hunk 3 above)
Comment (lmorelli986) [reply locales/en/index.yml]: changed in db325d10

File: locales/it/index.yml (hunk 4 above)
Comment (lmorelli986) [reply locales/it/index.yml]: changed in db325d10

File: ts/screens/onboarding/EmailValidateScreen.tsx (hunk 7 above)
Comment (lmorelli986) [reply ts/screens/onboarding/EmailValidateScreen.tsx]: changed in cdb554f8

File: ts/screens/onboarding/EmailValidateScreen.tsx (hunk 8 above)
Comment (lmorelli986) [reply ts/screens/onboarding/EmailValidateScreen.tsx]: changed in cdb554f8

File: ts/screens/onboarding/EmailValidateScreen.tsx (hunk 5 above)
Comment (lmorelli986) [reply ts/screens/onboarding/EmailValidateScreen.tsx]: changed in a05e3c15

File: ts/screens/onboarding/EmailValidateScreen.tsx (hunk 6 above)
Comment (lmorelli986) [reply ts/screens/onboarding/EmailValidateScreen.tsx]: removed in a05e3c15

File: ts/sagas/startup/checkAcknowledgedEmailSaga.ts (hunk 2 above)
Comment (lmorelli986) [reply ts/sagas/startup/checkAcknowledgedEmailSaga.ts]: question asked here #1308

Comment (lmorelli986) [reply ts/sagas/startup/checkAcknowledgedEmailSaga.ts]: questions answered here #1308

File: ts/screens/onboarding/EmailValidateScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+type Props = OwnProps & ReturnType<typeof mapDispatchToProps>;
+
+const styles = StyleSheet.create({
+  spacerLarge: { height: 40 }
+});
+
+const unavailableAlert = () => Alert.alert(I18n.t("global.notImplemented"));
+
+/**
+ * A screen to show if the fingerprint is supported to the user.
```
Comment (Undermaken) [inline ts/screens/onboarding/EmailValidateScreen.tsx]:
```suggestion
 * A screen as reminder to the user to validate his email address
```

**Comment by Undermaken** [comment]
@lmorelli986 could you solve the branch conflicts?

**Comment by lmorelli986** [comment]
waiting to #1316 to be integrated first, in order to merge with both [# 168246944,# 168247020] in a row

**Comment by Undermaken** [comment]
#1308 is merged ;)

**Comment by lmorelli986** [comment]
> Do you think it's possible to reduce the top margin between title and content?

markdown component is known to have some space-on-top issues. I removed the only padding I was adding in order to shrink content.

**Comment by lmorelli986** [comment]
Tried on iphone 5s and found that button labels are cut in italian so I slightly reduced fontSize

Was so:
Simulator Screen Shot - iPhone 5s - 2019-10-14 at 20 47 11

**Pull request merged**